Unable to generate migration Scripts in Azure DevOps Hosted Agent

<!-- Describe what isn't working as expected -->

2020-02-20T16:01:09.6186067Z ##[section]Starting: Build Migration Script
2020-02-20T16:01:09.6264348Z ==============================================================================
2020-02-20T16:01:09.6264395Z Task         : Command Line
2020-02-20T16:01:09.6264419Z Description  : Run a command line with arguments
2020-02-20T16:01:09.6264461Z Version      : 1.1.3
2020-02-20T16:01:09.6264485Z Author       : Microsoft Corporation
2020-02-20T16:01:09.6264508Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=613735)
2020-02-20T16:01:09.6264532Z ==============================================================================
2020-02-20T16:01:09.6271996Z ##[command]dotnet ef migrations script -p D:\a\1\s\PucAdmission -o D:\a\1\a\migrations\scripts.sql -i
2020-02-20T16:01:10.1898175Z It was not possible to find any compatible framework version
**2020-02-20T16:01:10.1908456Z The framework 'Microsoft.NETCore.App', version '3.1.2' was not found.**
2020-02-20T16:01:10.1908738Z   - The following frameworks were found:
2020-02-20T16:01:10.1908854Z       1.0.16 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1909260Z       1.1.13 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1909437Z       2.0.9 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1909564Z       2.1.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1909700Z       2.1.1 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1911328Z       2.1.2 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1911435Z       2.1.3 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1912270Z       2.1.4 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1913411Z       2.1.5 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1914594Z       2.1.6 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1915790Z       2.1.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1916727Z       2.1.8 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1917859Z       2.1.9 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1919102Z       2.1.10 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1920296Z       2.1.11 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1921411Z       2.1.12 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1922603Z       2.1.13 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1923699Z       2.1.14 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1924904Z       2.1.15 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1926315Z       2.2.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1927331Z       2.2.1 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1928171Z       2.2.2 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1929663Z       2.2.3 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1930893Z       2.2.4 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1933431Z       2.2.5 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1933599Z       2.2.6 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1933822Z       2.2.7 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1934691Z       2.2.8 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1937448Z       3.0.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1937864Z       3.0.1 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1937908Z       3.0.2 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1943887Z       3.1.0 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1943918Z       3.1.1 at [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
2020-02-20T16:01:10.1943935Z 
2020-02-20T16:01:10.1943987Z You can resolve the problem by installing the specified framework and/or SDK.
2020-02-20T16:01:10.1944006Z 
2020-02-20T16:01:10.1944032Z The specified framework can be found at:
2020-02-20T16:01:10.1944064Z   - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=3.1.2&arch=x64&rid=win10-x64
**2020-02-20T16:01:10.2286008Z ##[error]Process completed with exit code -2147450730.**
2020-02-20T16:01:10.2304011Z ##[section]Finishing: Build Migration Script



### Further technical details

Installed the Global tools before running this task using the below command:
`dotnet tool install --global dotnet-ef --version 3.1.2`

The installation succeeded but still this step fails.

The installed version of dotnet core is 3.1.0

Any possible workaround?

When I tried to install the SDK 3.1.2 Using the Use .Net Core Task I got the below error:

2020-02-20T17:40:14.6236374Z ##[section]Starting: Use .Net Core sdk 3.1.2
2020-02-20T17:40:14.6316226Z ==============================================================================
2020-02-20T17:40:14.6316280Z Task         : Use .NET Core
2020-02-20T17:40:14.6316307Z Description  : Acquires a specific version of the .NET Core SDK from the internet or the local cache and adds it to the PATH. Use this task to change the version of .NET Core used in subsequent tasks. Additionally provides proxy support.
2020-02-20T17:40:14.6316365Z Version      : 2.164.0
2020-02-20T17:40:14.6316391Z Author       : Microsoft Corporation
2020-02-20T17:40:14.6316421Z Help         : https://aka.ms/AA4xgy0
2020-02-20T17:40:14.6316464Z ==============================================================================
2020-02-20T17:40:15.0517216Z Tool to install: .NET Core sdk version 3.1.2.
2020-02-20T17:40:17.2485748Z No matching sdk version could be found for specified version: 3.1.2 Kindly note the preview versions are only considered in latest version searches if Include Preview Versions checkbox is checked.
2020-02-20T17:40:17.2486126Z Version 3.1.2 could not be found in its channel, will now search in adjacent channels.
2020-02-20T17:40:17.2488639Z Searching for version in channel 3.1
2020-02-20T17:40:18.8641343Z Searching for version in channel 3.0
2020-02-20T17:40:20.9094023Z ##[error]sdk version matching: 3.1.2 could not be found
2020-02-20T17:40:20.9157502Z ##[section]Finishing: Use .Net Core sdk 3.1.2

--- a/Directory.Build.targets
+++ b/Directory.Build.targets
@@ -5,7 +5,7 @@
     <!-- Use the latest version instead of the one bundled with the SDK -->
     <KnownFrameworkReference Update="Microsoft.NETCore.App">
       <LatestRuntimeFrameworkVersion Condition="'%(TargetFramework)' == '$(DefaultNetCoreTargetFramework)'">$(MicrosoftNETCoreAppRuntimeVersion)</LatestRuntimeFrameworkVersion>
-      <DefaultRuntimeFrameworkVersion Condition="'%(TargetFramework)' == '$(DefaultNetCoreTargetFramework)'">$(MicrosoftNETCoreAppRuntimeVersion)</DefaultRuntimeFrameworkVersion>
+      <DefaultRuntimeFrameworkVersion Condition="'%(TargetFramework)' == '$(DefaultNetCoreTargetFramework)' And '$(TargetLatestDotNetRuntime)' != 'false'">$(MicrosoftNETCoreAppRuntimeVersion)</DefaultRuntimeFrameworkVersion>
       <TargetingPackVersion Condition="'%(TargetFramework)' == '$(DefaultNetCoreTargetFramework)'">$(MicrosoftNETCoreAppRefPackageVersion)</TargetingPackVersion>
     </KnownFrameworkReference>
     <KnownFrameworkReference Update="NETStandard.Library">

--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -3,6 +3,7 @@
     <VersionPrefix>3.1.3</VersionPrefix>
     <PreReleaseVersionLabel>servicing</PreReleaseVersionLabel>
     <IncludeSourceRevisionInInformationalVersion>False</IncludeSourceRevisionInInformationalVersion>
+    <IsServicingBuild Condition="'$(PreReleaseVersionLabel)' == 'servicing'">true</IsServicingBuild>
     <!--
         When StabilizePackageVersion is set to 'true', this branch will produce stable outputs for 'Shipping' packages
     -->

